rocketmq-flume/rocketmq-flume-source/src/main/java/org/apache/rocketmq/flume/ng/source/RocketMQSource.java

```diff
@@ -119,31 +116,19 @@ protected Status doProcess() throws EventDeliveryException {
         Map<String, String> headers;
 
         try {
-            Set<MessageQueue> queues = consumer.fetchSubscribeMessageQueues(topic);
-            for (MessageQueue queue : queues) {
-                long offset = getMessageQueueOffset(queue);
-                PullResult pullResult = consumer.pull(queue, tag, offset, batchSize);
+            List<MessageExt> messageExts = consumer.poll();
+            for (MessageExt msg : messageExts) {
+                byte[] body = msg.getBody();
 
+                headers = new HashMap<>();
+                headers.put(HEADER_TOPIC_NAME, topic);
+                headers.put(HEADER_TAG_NAME, tag);
                 if (log.isDebugEnabled()) {
```

Comment:
You could remove this line

Reply:
Yeah, got it.

Reply:
Thanks, judging from the current code, especially the tests, it seems overly simplistic. I will test it as soon as possible and then consider merging

Reply:
@vongosling I think, may DefaultLitePullConsumer is better to adapter this scenario.